### DISCUSSION TITLE
Multiple connections between client and server

### DISCUSSION THREAD

**Discussion by GeeWee** [jchristn/WatsonWebsocket#66] [General]
Hello.

I'm not sure whether this is a feature request or a question.

I'm looking to implement _multiple, concurrent_ websocket connections between a client and a server. So that there can be `1-n` websockets concurrently between the same server and client.

Is this possible with WatsonWebsocket? My initial impression is _no_ - since the connections seem to be identified by `ipPort`, which is unique to one client, but doesn't leave any way to address multiple connections from the same client (where `ip` and `port` would both be identical)

If I'm missing something and this is possible - that's wonderful. Otherwise consider it a feature request :)

**Comment by jchristn** [comment]
Hi @GeeWee you can certainly create multiple instances of the ```WatsonWebsocketClient``` class.  Each instantiation would create a new connection (and thus a new port for each).  Cheers, Joel
